aws/data_source_aws_ecr_lifecycle_policy_document.go

```diff
@@ -0,0 +1,201 @@
+package aws
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+type EcrLifecyclePolicyDoc struct {
+	Rules []*EcrLifecyclePolicyStatement `json:"rules"`
+}
+
+type EcrLifecyclePolicyStatement struct {
+	RulePriority int                                     `json:"rulePriority,omitempty"`
+	Description  string                                  `json:"description,omitempty"`
+	Selection    EcrLifecyclePolicyStatementSelectionSet `json:"selection,omitempty"`
+	Action       EcrLifecyclePolicyAction                `json:"action"`
+}
+
+type EcrLifecyclePolicySelection struct {
+	TagStatus     string        `json:"tagStatus,omitempty"`
+	TagPrefixList []interface{} `json:"tagPrefixList,omitempty"`
+	CountType     string        `json:"countType,omitempty"`
+	CountUnit     string        `json:"countUnit,omitempty"`
+	CountNumber   int           `json:"countNumber,omitempty"`
+}
+
+type EcrLifecyclePolicyAction struct {
+	Type string `json:"type"`
+}
+
+type EcrLifecyclePolicyStatementSelectionSet EcrLifecyclePolicySelection
+
+func dataSourceAwsEcrLifecyclePolicyDocument() *schema.Resource {
+
+	return &schema.Resource{
+		Read: dataSourceAwsEcrLifecyclePolicyDocumentRead,
+
+		Schema: map[string]*schema.Schema{
+			"rule": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"priority": {
+							Type:     schema.TypeInt,
+							Required: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Required: false,
```

Comment:
Nit: `Required: false` is extraneous here and below.

Reply:
Both have been removed